velox/type/DecimalUtil.h

```diff
@@ -31,7 +32,7 @@ __attribute__((__no_sanitize__("signed-integer-overflow")))
 #endif
 inline int128_t
 mul(int128_t x, const int128_t y) {
-  return x * y;
+  return checkedMultiply<int128_t>(x, y);
```

Comment:
This should remain mul. We should probably use checkedMultiply in the function? Can you see what Java does?

Reply:
Also if it's really`checkedMultiply`, just use it without the wrapper, this one probably do not need annotation.

Reply:
Java throws exception:

```
resto:kart_test> select c0 * c0 from tint;

Query 20220826_000550_00008_82xud, FAILED, 1 node
Splits: 18 total, 0 done (0.00%)
131ms [0 rows, 0B] [0 rows/s, 0B/s]

Query 20220826_000550_00008_82xud failed: integer multiplication overflow: 2147483647 * 2147483647

```

Reply:
I will use checkedMultiply in the function and remove mul() for now.

Reply:
`mul` can be left there in other usage (like time with scale).  Just don't use it in the multiplication SQL function.